### ISSUE TITLE
Security vulnerability due to using com.nimbusds:nimbus-jose-jwt:5.1

### ISSUE THREAD

**Issue by sdenaci** [microsoft/react-native-code-push#2147]
### Steps to Reproduce
We have got a security warning flagged by NowSecure, our mobile security scanner. The test looks in the decompiled code of the binary for instances of static strings being used as cryptographic seeds. It advised us that code in our app is using a hardcoded byte array as a seed value for encryption. It traced this hardcoded byte array to `com.nimbusds:nimbus-jose-jwt:5.1`, (see stack trace below) which we traced to this library https://github.com/microsoft/react-native-code-push/blob/b9fb9fbe106dddb194940fa906364f80e4268c44/android/app/build.gradle#L30

### Expected Behavior
When initializing encryption, do not use a hardcoded byte array as a seed value. Instead, consider using dynamic values such as SecureRandom in order to create a secure initialization of the cryptography. The Findings Evidence table provides the instances where a hardcoded value was used.

see Android Developer Guidance and Code Samples for Secure Cryptography https://developer.android.com/guide/topics/security/cryptography

A suggested fix might be to use a different library or encourage nimbus to change how they are seeding encryption.

### Actual Behavior
The application is using static values as seeds for encryption. This means that any attacker who sees those static seeds will be able to easily decrypt that data. If the data being encrypted is sensitive information, the attacker will be able to see and exploit the information.

```
  STACK TRACE AND/OR SCREENSHOTS
```

 {
  "location": "Lcom/nimbusds/jose/crypto/AESGCMKW;,decryptCEK,(Ljavax/crypto/SecretKey;[BLcom/nimbusds/jose/crypto/AuthenticatedCipherText;ILjava/security/Provider;)Ljavax/crypto/SecretKey;",
  "hardcoded_value": "Lcom/nimbusds/jose/crypto/AESGCM;,decrypt,(Ljavax/crypto/SecretKey;[B[B[B[BLjava/security/Provider;)[B",
  "instance_type": "Ljavax/crypto/spec/SecretKeySpec;"
}

### Environment

* react-native-code-push version: 7.0.3
* react-native version: 0.64.2
* iOS/Android/Windows version: android 29
* Does this reproduce on a debug build or release build?: release
* Does this reproduce on a simulator, or only on a physical device?: both

(The more info the faster we will be able to address it!)

**Comment by owjsub** [comment]
https://github.com/microsoft/react-native-code-push/blob/36cd93770d761b3db1c9d2f379060a50bf8f1464/android/app/build.gradle#L30

https://nvd.nist.gov/vuln/detail/CVE-2019-17195

If someone could update this dependency version to at least `7.9`, that would be awesome!

**Comment by ganesh-papola** [comment]
is there any update? are we upgrading `com.nimbusds:nimbus-jose-jwt`?

**Comment by nikolal** [comment]
We are also informed about this by a specialised security agency as a high vulnerability risk:

`com/nimbusds/jose/crypto/AESCBC.java`

`com/nimbusds/jose/jca/JCASupport.java`

```
It is observed that the App uses the encryption mode CBC with PKCS5/PKCS7 padding.
Padding oracle attacks are a type of cryptographic attack that exploits the
implementation of encryption algorithms that use padding.
```

Any plans of addressing this issue?

**Comment by kennethnokman** [comment]
Any plans from @msftgits to fix it?

**Comment by nikolal** [comment]
The issue is fixed in v8.2.2.
